ptf

```diff
@@ -525,6 +525,11 @@ def load_test_modules(config):
 
     for root, dirs, filenames in os.walk(config["test_dir"]):
         pyfiles = fnmatch.filter(filenames, "[!.]*.py")
+
+        # guarantee that files will be visited in the same order every time tests are loaded
+        pyfiles.sort()
+        dirs.sort()
```

Comment:
It is used by `os.walk` itself on every iteration. https://docs.python.org/3/library/os.html#os.walk

> When topdown is True, **the caller can modify the dirnames list in-place** (perhaps using [del](https://docs.python.org/3/reference/simple_stmts.html#del) or slice assignment), and [walk()](https://docs.python.org/3/library/os.html#os.walk) will only recurse into the subdirectories whose names remain in dirnames; **this can be used to** prune the search, **impose a specific order of visiting**, or even to inform [walk()](https://docs.python.org/3/library/os.html#os.walk) about directories the caller creates or renames before it resumes [walk()](https://docs.python.org/3/library/os.html#os.walk) again. Modifying dirnames when topdown is False has no effect on the behavior of the walk, because in bottom-up mode the directories in dirnames are generated before dirpath itself is generated.

Reply:
Thanks for the clarification!